src/components/PostsList.tsx

```diff
@@ -1,54 +1,62 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
 
 import classNames from 'classnames';
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Post } from '../types/Post';
+import { useAppSelector } from '../app/hooks';
+import { useDispatch } from 'react-redux';
+import { setActivePost } from '../features/postsSlice';
 
 type Props = {
   posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
 };
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
-
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
-
-      <tbody>
-        {posts.map(post => (
-          <tr key={post.id} data-cy="Post">
-            <td data-cy="PostId">{post.id}</td>
-            <td data-cy="PostTitle">{post.title}</td>
-            <td className="has-text-right is-vcentered">
-              <button
-                type="button"
-                data-cy="PostButton"
-                className={classNames('button', 'is-link', {
-                  'is-light': post.id !== selectedPostId,
-                })}
-                onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
-                }}
-              >
-                {post.id === selectedPostId ? 'Close' : 'Open'}
-              </button>
-            </td>
+export const PostsList: React.FC<Props> = ({ posts }) => {
+  const { activePost } = useAppSelector(state => state.posts);
+  const dispatch = useDispatch();
+
+  const handleActivePost = (post: Post) => {
+    dispatch(setActivePost(post.id === activePost?.id ? null : post));
+  };
+
+  useEffect(() => {
+    dispatch(setActivePost(null));
+  }, []);
```

Comment:
React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array